### PR TITLE
Pull and save Docker images in parallel

### PR DESCRIPTION
**Pull request by praseodym**
On my machine, this change speeds up `kind build node-image --type bazel` by a few seconds:
* Images not in local Docker cache: 36.10s => 29.52s
* Images already in local Docker cache: 23.68s => 20.03s

### CODE REVIEW

File: pkg/build/node/node.go [hunk 1]
```diff
@@ -446,23 +449,36 @@ func (c *BuildContext) prePullImages(dir, containerID string) error {
...
-				return err
-			}
-			// TODO(bentheelder): generate a friendlier name
-			pullName := fmt.Sprintf("%d.tar", i)
-			pullTo := path.Join(imagesDir, pullName)
-			err = docker.Save(image, pullTo)
-			if err != nil {
-				return err
+		i, image := i, image // https://golang.org/doc/faq#closures_and_goroutines
+		g.Go(func() error {
```
Comment (aojea) [inline pkg/build/node/node.go]: what about using current concurrent functions https://github.com/kubernetes-sigs/kind/blob/master/pkg/concurrent/concurrent.go?

There are some examples in the code of using it in Ben´s commit https://github.com/kubernetes-sigs/kind/commit/110dc82f093cdcb7835a01ba85c1ce714083f434

Comment (praseodym) [reply pkg/build/node/node.go]: Thanks, didn't know about these. Changed! :slightly_smiling_face:

**Comment by aojea** [comment]
Thanks
/ok-to-test

**Comment by aojea** [comment]
SGTM
we should wait for @BenTheElder review

File: pkg/build/node/node.go (hunk 1 above)
Comment (BenTheElder) [reply pkg/build/node/node.go]: this other library is really neat though! will need to revisit this sometime, thanks for the pointer :-)

**Pull request merged**